### PR TITLE
Add support for defining custom HTTP error codes

### PR DESCRIPTION
**Pull request by osterman**
## what
* allow `custom-error-codes` to be defined in `values.yaml`

## why
* some applications might already generate nice status pages
* other times we might want to intercept more status codes (e.g. `500` and `502`)

## who
@goruha

### CODE REVIEW

File: incubator/nginx-ingress/templates/configmap.yaml [hunk 1]
```diff
@@ -6,6 +6,6 @@ metadata:
     k8s-addon: ingress-nginx.addons.k8s.io
 data:
   use-proxy-protocol: "true"
-  custom-http-errors: 404,502,503,504
-
-
+{{- if .Values.customHttpErrors }}
+  custom-http-errors: "{{ range $index, $element := .Values.customHttpErrors}}{{if $index}}, {{end}}{{$element}}{{end}}"
```
Comment (goruha) [inline incubator/nginx-ingress/templates/configmap.yaml]: Use Join function instead of range

https://github.com/Masterminds/sprig#string-slice-functions

Comment (osterman) [inline incubator/nginx-ingress/templates/configmap.yaml]: https://kubernetes.slack.com/archives/helm/p1488051682014608

Comment (osterman) [inline incubator/nginx-ingress/templates/configmap.yaml]: https://cloudposse.slack.com/archives/ops/p1488051184000493

Comment (osterman) [inline incubator/nginx-ingress/templates/configmap.yaml]: Haha =) i tried - remember, this is what was making me go mad.

File: incubator/nginx-ingress/templates/configmap.yaml [hunk 2]
```diff
@@ -6,6 +6,14 @@ metadata:
     k8s-addon: ingress-nginx.addons.k8s.io
 data:
   use-proxy-protocol: "true"
-  custom-http-errors: 404,502,503,504
-
-
+  custom-http-errors: "
+    {{- range $index, $element := .Values.customHttpErrors.client -}}
```
Comment (goruha) [inline incubator/nginx-ingress/templates/configmap.yaml]: I know this looks like madness.
But go template is crap in arrays and lists, that is the best I can do.

**Pull request merged**